### PR TITLE
Added predefined resolutions

### PR DESCRIPTION
**Pull request by rutgersmit**
Added a list with predefined resolutions.
The list is populated by a textfile resolutions.txt and is read when the application starts.

### CODE REVIEW

**Comment by benabraham** [comment]
@tom-englert This would be awesome. Could you please merge it?

File: src/RegionToShare/MainWindow.xaml [hunk 1]
```diff
@@ -1,19 +1,27 @@
-﻿<Window x:Class="RegionToShare.MainWindow"
+﻿<Window
         xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        x:Class="RegionToShare.MainWindow"
```
[the remark this answers is not in the document]
Comment (rutgersmit) [reply src/RegionToShare/MainWindow.xaml:4]: Visual Studio did it during migration

[the remark this answers is not in the document]
Comment (tom-englert) [reply src/RegionToShare/MainWindow.xaml:4]: But you committed it 😏

[the remark this answers is not in the document]
Comment (rutgersmit) [reply src/RegionToShare/MainWindow.xaml:4]: The 'Window' root element requires a x:Class attribute to support event handlers in the XAML file.

File: src/RegionToShare/MainWindow.xaml.cs [hunk 2]
```diff
@@ -17,9 +19,19 @@ public partial class MainWindow
...
 
+    public ObservableCollection<string> Resolutions { get; set; }
+
     public MainWindow()
     {
         InitializeComponent();
+        Resolutions = new ObservableCollection<string>();
+        DataContext = this;
+
+        if (!File.Exists("resolutions.txt"))
```
[the remark this answers is not in the document]
Comment (rutgersmit) [reply src/RegionToShare/MainWindow.xaml.cs]: Rewritten that part, file is created in a dir 'RegionToShare' in the ApplicationData path

File: src/RegionToShare/MainWindow.xaml.cs [hunk 3]
```diff
@@ -151,4 +164,20 @@ public void SendToBack()
...
     }
+
+
+    private void lbResolutions_SelectionChanged(object sender, System.Windows.Controls.SelectionChangedEventArgs e)
+    {
+        var resolution = e.AddedItems[0].ToString();
+        try
+        {
+            // user can added garbage to txt file
+            Width = Convert.ToDouble(resolution.Split('x')[0]);
```
[the remark this answers is not in the document]
Comment (rutgersmit) [reply src/RegionToShare/MainWindow.xaml.cs]: Ok.

File: src/RegionToShare/MainWindow.xaml [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
-              Fill="{Binding Foreground, RelativeSource={RelativeSource FindAncestor, AncestorType=Window}}"
+    <Grid DataContext="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}}">
+        <Grid x:Name="InfoArea" Background="{StaticResource HatchBrush}">
+            <TextBlock x:Name="Extend" Background="Black" VerticalAlignment="Bottom" HorizontalAlignment="Right" Padding="5,2"/>
+            <Grid VerticalAlignment="Center" HorizontalAlignment="Center" Grid.Row="3">
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="1*" />
+                    <RowDefinition Height="2*" />
+                </Grid.RowDefinitions>
+                <ListBox Grid.Row="0" x:Name="lbResolutions" ItemsSource="{Binding Resolutions}" Background="Black" Foreground="Green" BorderBrush="Black" Margin="0,0,0,25" FontSize="22" FontWeight="Bold" SelectionChanged="lbResolutions_SelectionChanged" />
```
[the remark this answers is not in the document]
Comment (tom-englert) [reply src/RegionToShare/MainWindow.xaml:15]: - Select first size
- manually resize the window
- Try to select first size again

[the remark this answers is not in the document]
Comment (rutgersmit) [reply src/RegionToShare/MainWindow.xaml:15]: Ah, ok. Fixed that now. Not via a two way binding but just via resetting the selectedindex. Hey, it works right?

File: src/RegionToShare/MainWindow.xaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tom-englert) [reply src/RegionToShare/MainWindow.xaml:4]: Yes, and this should be the first entry for consistency with all other XAML.

File: src/RegionToShare/MainWindow.xaml.cs [hunk 5]
```diff
@@ -17,9 +19,30 @@ public partial class MainWindow
...
         InitializeComponent();
+        Resolutions = new ObservableCollection<string>();
+        DataContext = this;
+
+        LoadResolutions();
+    }
+
+    private void LoadResolutions() 
+    {
+        var userDataDirPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "\\RegionToShare";
```
Comment (tom-englert) [inline src/RegionToShare/MainWindow.xaml.cs:35]: Never manually mangle with path names, use Path.Combine

**Comment by tom-englert** [comment]
Superseded by a317ae4. Can you please check out the latest version from and have a quick test?

**Comment by benabraham** [comment]
Thank you!

**Pull request closed without merging**